### PR TITLE
More linters and better circle-ci config

### PR DESCRIPTION
**Pull request by dnephin**
Add the last 2 linters.

`lll` checks line length. 200 is rather long, but it's a start
`interfacer` checks for when an unnecessarily large interface is used.

I've also updated the circleci config to run each of the steps in parallel. This cuts down the build time to just the slowest step, which is currently `cross`. With #26 cross is going to get even slower.

With this change we can also download binaries for each build: https://circleci.com/gh/docker/cli/51#artifacts/containers/1

### CODE REVIEW

File: cli/command/container/cmd.go [hunk 1]
```diff
@@ -7,12 +7,13 @@ import (
 )
 
 // NewContainerCommand returns a cobra command for `container` subcommands
+// nolint: interfacer
 func NewContainerCommand(dockerCli *command.DockerCli) *cobra.Command {
```
Comment (vdemeester) [inline cli/command/container/cmd.go]: @dnephin why not this one for `command.Cli` ?

File: cli/command/image/trust.go [hunk 2]
```diff
@@ -42,7 +42,7 @@ func trustedPush(ctx context.Context, cli *command.DockerCli, repoInfo *registry
 
 // PushTrustedReference pushes a canonical reference to the trust server.
 // nolint: gocyclo
-func PushTrustedReference(cli *command.DockerCli, repoInfo *registry.RepositoryInfo, ref reference.Named, authConfig types.AuthConfig, in io.Reader) error {
+func PushTrustedReference(streams command.Streams, repoInfo *registry.RepositoryInfo, ref reference.Named, authConfig types.AuthConfig, in io.Reader) error {
```
Comment (vdemeester) [inline cli/command/image/trust.go]: 😻

File: cli/command/network/cmd.go [hunk 3]
```diff
@@ -8,12 +8,13 @@ import (
 )
 
 // NewNetworkCommand returns a cobra command for `network` subcommands
+// nolint: interfacer
 func NewNetworkCommand(dockerCli *command.DockerCli) *cobra.Command {
```
Comment (vdemeester) [inline cli/command/network/cmd.go]: Same here.

File: cli/command/plugin/cmd.go [hunk 4]
```diff
@@ -7,12 +7,13 @@ import (
 )
 
 // NewPluginCommand returns a cobra command for `plugin` subcommands
+// nolint: interfacer
 func NewPluginCommand(dockerCli *command.DockerCli) *cobra.Command {
```
Comment (vdemeester) [inline cli/command/plugin/cmd.go]: and here 👼

File: cli/command/secret/cmd.go [hunk 5]
```diff
@@ -8,12 +8,13 @@ import (
 )
 
 // NewSecretCommand returns a cobra command for `secret` subcommands
+// nolint: interfacer
 func NewSecretCommand(dockerCli *command.DockerCli) *cobra.Command {
```
Comment (vdemeester) [inline cli/command/secret/cmd.go]: here too, and below 😛

File: cli/command/container/cmd.go (hunk 1 above)
Comment (dnephin) [inline cli/command/container/cmd.go]: There are some auth methods that don't exist on the interface yet. And if I change this I have to change everything in the `container/` package, so I wanted to delay that change.

Most `// nolint` should be considered the equivalent of a `// TODO: fix this`

**Comment by dnephin** [comment]
Sure, I'm fine to wait

**Pull request merged**